src/logger/index.mjs

```diff
@@ -0,0 +1,57 @@
+// @ts-check
+
+'use strict';
+
+import { createLogger } from './logger.mjs';
+import { transports } from './transports/index.mjs';
+
+/**
+ * @typedef {ReturnType<typeof createLogger>} LoggerInstance
+ */
+
+/**
+ * Singleton logger instance for consistent logging across the app.
+ */
+export const Logger = (() => {
```

Comment:
I don't think you need to wrap this whole thing as a function. You can just move this code to top level.

Or at least, do:

```
export const Logger = () => {}; // <- raw export of the factory.

export default Logger(); // <- default export of the singleton

Reply:
What about the transport param in the `init` method?